### PR TITLE
[FIX] account, purchase, sale: included taxes

### PR DESCRIPTION
**Pull request by simongoffin**
Forward port of 503820acb6d82a85c0c49ac26f7e7f8cd73851dd from 8.0

opw:652310

### CODE REVIEW

File: addons/account/models/account.py [hunk 1]
```diff
@@ -679,7 +679,7 @@ def compute_all(self, cr, uid, ids, price_unit, currency_id=None, quantity=1.0,
     def _fix_tax_included_price(self, price, prod_taxes, line_taxes):
         """Subtract tax amount from price when corresponding "price included" taxes do not apply"""
         # FIXME get currency in param?
-        incl_tax = prod_taxes.filtered(lambda tax: tax.id not in line_taxes and tax.price_include)
+        incl_tax = prod_taxes.filtered(lambda tax: tax.id not in line_taxes.ids and tax.price_include)
```
Comment (nim-odoo) [inline addons/account/models/account.py]: `incl_tax = prod_taxes.filtered(lambda tax: tax not in line_taxes and tax.price_include)`

File: addons/purchase/purchase.py [hunk 2]
```diff
@@ -584,6 +584,7 @@ def onchange_product_id(self):
 
         fpos = self.order_id.fiscal_position_id
         self.taxes_id = fpos.map_tax(self.product_id.supplier_taxes_id)
+        self.price_unit = self.env['account.tax']._fix_tax_included_price(self.price_unit, self.product_id.supplier_taxes_id, self.taxes_id)
```
Comment (nim-odoo) [inline addons/purchase/purchase.py]: You should replace:
`price_unit = seller.price if seller else 0.0`
With:
`price_unit = self.env['account.tax']._fix_tax_included_price(seller.price, self.product_id.supplier_taxes_id, self.taxes_id) if seller else 0.0`

This appears 4 times in purchase.py

**Comment by nim-odoo** [comment]
@fwi-odoo : this might be useful for eBay as well

**Comment by simongoffin** [comment]
@odony

**Comment by odony** [comment]
Any idea what happened with the tests in 503820a ?

**Comment by simongoffin** [comment]
@odony 
Alright!
All the tests are there now.

**Pull request closed without merging**